### PR TITLE
Job to validate stitch tables

### PR DESCRIPTION
**Pull request by pwnage101**
DENG-360

### CODE REVIEW

File: dataeng/jobs/analytics/SnowflakeValidateStitch.groovy [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+class SnowflakeValidateStitch {
+    public static def job = { dslFactory, allVars ->
+            dslFactory.job('snowflake-validate-stitch') {
+
+                description(
+                    'Validate a application database tables loaded by Stitch by comparing them against the same ' +
+                    'tables loaded by Sqoop.  This compares only tables that exist in both sets, and only the last ' +
+                    '10 days of changed rows.'
+                )
```
Comment (pwnage101) [inline dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:18]: Look, a description!

File: dataeng/jobs/analytics/SnowflakeValidateStitch.groovy [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                'tables loaded by Sqoop.  This compares only tables that exist in both sets, and only the last ' +
+                '10 days of changed rows.'
+            )
+
+            logRotator common_log_rotator(allVars)
+            parameters secure_scm_parameters(allVars)
+            parameters {
+                stringParam('ANALYTICS_TOOLS_URL', allVars.get('ANALYTICS_TOOLS_URL'), 'URL for the analytics tools repo.')
+                stringParam('ANALYTICS_TOOLS_BRANCH', allVars.get('ANALYTICS_TOOLS_BRANCH'), 'Branch of analytics tools repo to use.')
+                stringParam('APP_NAME', '', 'Application name of tables to validate.')
```
Comment (doctoryes) [inline dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:25]: Is the intent here to have a single Jenkins job that checks all four app DBs? Or four separate jobs? Where does `APP_NAME` come from here?

Comment (doctoryes) [reply dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:25]: Nm - just saw https://github.com/edx/jenkins-job-dsl/pull/1075

File: dataeng/jobs/analytics/SnowflakeValidateStitch.groovy [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm
+import static org.edx.jenkins.dsl.AnalyticsConstants.secure_scm_parameters
+
+
+class SnowflakeValidateStitch {
+    public static def job = { dslFactory, allVars ->
+        dslFactory.job('snowflake-validate-stitch') {
+
+            description(
+                'Validate a application database tables loaded by Stitch by comparing them against the same ' +
```
Comment (brianhw) [inline dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:15]: nit: "Validate a application" => "Validate application"

File: dataeng/jobs/analytics/SnowflakeValidateStitch.groovy [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                stringParam('APP_NAME', '', 'Application name of tables to validate.')
+                stringParam('SQOOP_START_TIME', '', 'Application name of tables to validate.')
+            }
+            environmentVariables {
+                env('KEY_PATH', allVars.get('KEY_PATH'))
+                env('PASSPHRASE_PATH', allVars.get('PASSPHRASE_PATH'))
+                env('USER', allVars.get('USER'))
+                env('ACCOUNT', allVars.get('ACCOUNT'))
+            }
+            logRotator common_log_rotator(allVars)
```
Comment (brianhw) [inline dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:34]: looks like this appears twice....

File: dataeng/resources/snowflake-validate-stitch.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -ex
+
+# Calculate the start of the validation window as 10 days prior to the end of the window.
+BEGIN_TIME=$(date -d "${SQOOP_START_TIME} - 10 days")
```
Comment (brianhw) [inline dataeng/resources/snowflake-validate-stitch.sh:5]: nit: maybe call it COMPARISON_START_TIME ?  You could even define a COMPARISON_END_TIME too, and include that in the (already useful) comment.

File: dataeng/jobs/analytics/SnowflakeValidateStitch.groovy (hunk 2 above)
Comment (pwnage101) [reply dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:25]: yea, the intent is a single jenkins job to check all four app dbs.  That actually reminds me, I should update the build label to include the app name.

Comment (doctoryes) [reply dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:24]: If a single invocation of the `stitch_vs_sqoop_validation.py` is supposed to validate all four schemas, it'll need to be `--schema CREDENTIALS --schema LMS --schema ECOMMERCE --schema DISCOVERY`.

Comment (doctoryes) [reply dataeng/jobs/analytics/SnowflakeValidateStitch.groovy:24]: That is, unless I change the script to work as `--schema CREDENTIALS LMS ECOMMERCE DISCOVERY`.

**Comment by pwnage101** [comment]
> Is the invocation of the validation script correct? Is `APP_NAME` intended to be a space-separated list of application DBs?

No, it's intended to be a single app name, not a list.  IIRC the python script will take a list, but that doesn't work anyway because different apps will need different start and end times.  This will produce one single jenkins job called `snowflake-validate-stitch` which will in turn have a total of 4 upstream jobs, one for each snowflake schema loading job.

**Pull request merged**